src/dmd/root/outbuffer.d

```diff
@@ -510,6 +539,80 @@ struct OutBuffer
             writeByte(0);
         return extractData();
     }
+
+    extern(D) bool moveToFile(const char* filename)
+    {
+        import dmd.root.file;
+        if (data == FileMapping!(const ubyte)(filename)[])
+        {
+            // Contents is the same, just touch the file
+            File.touch(filename);
+            // Carefully discard all data from this buffer.
+            if (!fileMapping.active)
+            {
+                destroy();
+                return true;
+            }
+            // Truncate file to zero so unflushed buffers are not flushed unnecessarily.
+            fileMapping.resize(0);
+            auto tmpfilename = fileMapping.filename;
+            fileMapping.close();
+            data = null;
+            offset = 0;
+            // Get rid of the underlying temp file.
+
+            version(Posix)
+            {
+                import core.sys.posix.unistd;
+                if (unlink(tmpfilename) != 0)
+                {
+                    perror("unlink");
+                    return false;
+                }
+            }
+            else version(Windows)
+            {
+                import core.sys.windows.winbase;
+                if (DeleteFileA(filename) == 0)
+                {
+                    assert(0);
```

Comment:
Maybe a better fatal error message than that!

Reply:
Shouldn't assert on environmental errors.

Reply:
Ooops, all `assert(0)`s were placeholders waiting for actual meaningful error messages... will fix tomorrow.